Mathlib/NumberTheory/Cyclotomic/Basic.lean

```diff
@@ -884,3 +888,112 @@ instance IsSepClosedOfCharZero.isCyclotomicExtension [CharZero K] :
 alias IsAlgClosedOfCharZero.isCyclotomicExtension := IsSepClosedOfCharZero.isCyclotomicExtension
 
 end IsSepClosed
+
+section Subalgebra
+
+theorem IsCyclotomicExtension.empty_eq (C : Subalgebra A B) [hC : IsCyclotomicExtension ∅ A C] :
+    C = ⊥ := by
+  simpa [Subalgebra.range_val] using
+    congr_arg (Subalgebra.map C.val) (IsCyclotomicExtension.empty A C).symm
```

Comment:
I think if we add the `Subsingleton` instance suggested [here](https://github.com/leanprover-community/mathlib4/pull/29665/files#r2386174256) then we can drop this theorem since it will just be a special case of `Subsingleton.elim`.